client/vehicle.lua

```diff
@@ -0,0 +1,245 @@
+local config = require 'config.client'
+local currentindicatorL, currentindicatorR = false, false
+local getEntitySpeed = GetEntitySpeed
+local getVehicleFuelLevel = GetVehicleFuelLevel
+local getIsVehicleEngineRunning = GetIsVehicleEngineRunning
+local getVehicleLightsState = GetVehicleLightsState
+local getVehicleTrailerVehicle = GetVehicleTrailerVehicle
+local getVehicleIndicatorLights = GetVehicleIndicatorLights
+local isMinimapRendering = IsMinimapRendering
+local isControlJustPressed = IsControlJustPressed
+local setVehicleIndicatorLights = SetVehicleIndicatorLights
+local sendNUIMessage = SendNUIMessage
+
+local function vehiclehudloop()
+    local currentindicators = getVehicleIndicatorLights(cache.vehicle)
+    local indl = currentindicators == 1 or currentindicators == 3
+    local indr = currentindicators == 2 or currentindicators == 3
+    local warning = false
+
+    CreateThread(function()
+        local sleep
+        while cache.seat == -1 do
+            sleep = 1000
+                if getIsVehicleEngineRunning(cache.vehicle) and isMinimapRendering() then
+                local HasTrailer, Trailer = getVehicleTrailerVehicle(cache.vehicle)
+                if isControlJustPressed(1, 174) then     -- <- is pressed
+                    indl = warning or not indl
+                    indr = indr and false

```

Comment:
Use the warning variable here as well to make sure it just doesn't turn off the indicator when those lights are on, if the warning variable is false, make indr just false, no need for that extra check